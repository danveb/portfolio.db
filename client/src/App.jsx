import { useState } from "react"; 
import { Navbar, Menu, Hero, About, Projects, Contact, Footer } from "./components"; 
import "./App.scss"; 

const App = () => {
    const [menuOpen, setMenuOpen] = useState(false)
    return (
        <div className="app">
            <Navbar menuOpen={menuOpen} setMenuOpen={setMenuOpen} />
            <Menu menuOpen={menuOpen} setMenuOpen={setMenuOpen} />
            <div className="sections">
                <Hero />
                <About />
                <Projects />
                <Contact />
                <Footer />
            </div>
        </div>
    )
}

export default App