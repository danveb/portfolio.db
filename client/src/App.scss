.app {
    height: 100vh; 

    .sections {
        width: 100%; 
        height: calc(100vh - 50px); 
        position: relative; 
        top: 50px; 
        scroll-behavior: smooth; 
        scrollbar-width: none; 
        &::-webkit-scrollbar {
            display: none; 
        }

        > * {
            width: 100vw; 
            scroll-snap-align: start; 
        }
    }
}