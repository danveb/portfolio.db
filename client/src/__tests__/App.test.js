import { render } from "@testing-library/react"; 
import App from "../App"; 

describe("App component", () => {
    // smoke test
    test("renders without crashing", () => {
        render(<App />); 
    });

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(<App />); 
        expect(asFragment()).toMatchSnapshot(); 
    });
});