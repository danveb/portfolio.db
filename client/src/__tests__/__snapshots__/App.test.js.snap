// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="navbar false"
    >
      <div
        class="navbar-left"
      >
        <a
          class="logo"
          href="#intro"
        >
          db
          <span>
            .
          </span>
        </a>
      </div>
      <div
        class="navbar-right"
      >
        <ul
          class="navbar-links"
        >
          <li>
            <a
              href="#about"
            >
              about
            </a>
          </li>
          <li>
            <a
              href="#projects"
            >
              projects
            </a>
          </li>
          <li>
            <a
              href="#contact"
            >
              contact
            </a>
          </li>
          <li>
            <div>
              <a
                href="https://drive.google.com/file/d/1N0VvKMKcAfogYAo-3gB3YONRT540EPLr/view?usp=sharing"
                rel="noopener noreferrer"
                target="_blank"
              >
                resume
              </a>
            </div>
          </li>
        </ul>
        <div
          class="hamburger"
        >
          <span
            class="line1"
          />
          <span
            class="line2"
          />
          <span
            class="line3"
          />
        </div>
      </div>
    </div>
    <div
      class="menu false"
    >
      <ul>
        <li>
          <a
            href="#about"
          >
            about
          </a>
        </li>
        <li>
          <a
            href="#projects"
          >
            projects
          </a>
        </li>
        <li>
          <a
            href="#contact"
          >
            contact
          </a>
        </li>
        <li>
          <div>
            <a
              href="https://drive.google.com/file/d/1N0VvKMKcAfogYAo-3gB3YONRT540EPLr/view?usp=sharing"
              rel="noopener noreferrer"
              target="_blank"
            >
              resume
            </a>
          </div>
        </li>
      </ul>
      <ul>
        <li>
          <a
            class="social"
            href="https://www.github.com/danveb"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            class="social"
            href="https://www.linkedin.com/in/daniel-eze-bae"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkedInIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            class="social"
            href="mailto:danibae91@gmail.com"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AlternateEmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10h5v-2h-5c-4.34 0-8-3.66-8-8s3.66-8 8-8 8 3.66 8 8v1.43c0 .79-.71 1.57-1.5 1.57s-1.5-.78-1.5-1.57V12c0-2.76-2.24-5-5-5s-5 2.24-5 5 2.24 5 5 5c1.38 0 2.64-.56 3.54-1.47.65.89 1.77 1.47 2.96 1.47 1.97 0 3.5-1.6 3.5-3.57V12c0-5.52-4.48-10-10-10zm0 13c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="sections"
    >
      <div
        class="hero"
        id="intro"
      >
        <div
          class="hero__wrapper"
        >
          <div
            class="hero__headline"
          >
            <h1>
              Hi, I'm Danny
              <span>
                .
              </span>
              <span>
                Frontend Developer
                <span>
                  .
                </span>
              </span>
            </h1>
          </div>
          <div
            class="hero__introduction"
          >
            <p>
              I'm passionate about finding ways to efficiently solve complex problems and make a positive difference for today and for tomorrow.
            </p>
          </div>
          <div
            class="hero__btn"
          >
            <a
              class="cta"
              href="https://www.linkedin.com/in/daniel-eze-bae"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="hover"
              />
              <span>
                Let's Work Together
              </span>
            </a>
          </div>
          <div
            class="hero__laptop"
          >
            <img
              alt="laptop hero"
              src="sail-ly.png"
            />
          </div>
          <div
            class="scroll-arrow"
          >
            <svg
              height="30px"
              width="30px"
            >
              <path
                d="M2.000,5.000 L15.000,18.000 L28.000,5.000 "
                fill="none"
                stroke="#ffffff"
                stroke-width="2px"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="about"
        id="about"
      >
        <div
          class="about__wrapper"
        >
          <div
            class="about__main"
          >
            <div
              class="about__title"
            >
              <h3>
                My Story
                <span>
                  .
                </span>
              </h3>
            </div>
            <div
              class="about__img"
            >
              <img
                alt="profile pic bw"
                src="profile-bw.png"
              />
            </div>
            <div
              class="about__story"
            >
              <p>
                My journey started in 2020 when I wrote my first line of code, 
                <span>
                  Hello World
                </span>
                 on an HTML document. This simple, yet powerful line of code was the catalyst to learning web development (courses on HTML, CSS & JavaScript) and thus building my first static website.
              </p>
              <p>
                Fast-forward to today, most of my web development work is with React. I design and develop web apps that provide a rich user interface/experience. I focus on creating clean, functional, scalable, and testable apps.
              </p>
              <p>
                On weekends I spend time tufting cool rugs, crafting home-made boba tea with friends and dreaming about soccer.
              </p>
            </div>
            <div
              class="about__skills"
            >
              <ul>
                <li>
                  <img
                    alt="html logo"
                    src="html5.png"
                  />
                </li>
                <li>
                  <img
                    alt="css logo"
                    src="css.png"
                  />
                </li>
                <li>
                  <img
                    alt="javascript logo"
                    src="js.png"
                  />
                </li>
                <li>
                  <img
                    alt="react logo"
                    src="react.png"
                  />
                </li>
                <li>
                  <img
                    alt="sass logo"
                    src="sass.png"
                  />
                </li>
                <li>
                  <img
                    alt="express logo"
                    src="express.png"
                  />
                </li>
                <li>
                  <img
                    alt="psql logo"
                    src="psql.png"
                  />
                </li>
                <li>
                  <img
                    alt="mongoDB logo"
                    src="mongoDB.png"
                  />
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="projects"
        id="projects"
      >
        <div
          class="projects__wrapper"
        >
          <div
            class="projects__main"
          >
            <div
              class="projects__title"
            >
              <h3>
                Projects
                <span>
                  .
                </span>
              </h3>
            </div>
            <div
              class="projects__cards"
            >
              <div
                class="card__container"
              >
                <div
                  class="image__container"
                >
                  <img
                    alt="portfolio project showcase"
                    src="sail-ly.png"
                  />
                </div>
                <div
                  class="card__content"
                >
                  <div
                    class="card__title"
                  >
                    <h1>
                      sail.ly
                    </h1>
                  </div>
                  <div
                    class="card__body"
                  >
                    <p>
                      Dream in open waters. SAIL.LY is a web app for beginner-to-experienced sailors that want to enjoy a day out on the water. App allows users to check on their favorite sailing clubs for sailing conditions based on real-time weather data, integrated with Weather API.
                    </p>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      React
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      CSS
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Node
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Express
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      PostgreSQL
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Weather API
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      MapboxGL API
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Netlify
                    </button>
                  </div>
                  <div
                    class="card__footer"
                  >
                    <a
                      href="https://sail-ly.netlify.app"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      Live Site
                    </a>
                    <a
                      href="https://github.com/danveb/sail.ly"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      GitHub Repo
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="card__container"
              >
                <div
                  class="image__container"
                >
                  <img
                    alt="portfolio project showcase"
                    src="bobaLife.png"
                  />
                </div>
                <div
                  class="card__content"
                >
                  <div
                    class="card__title"
                  >
                    <h1>
                      bobaLife
                    </h1>
                  </div>
                  <div
                    class="card__body"
                  >
                    <p>
                      Who enjoys sipping an ice-cold boba milk tea at any time of the day? bobaLife is an e-commerce web app designed for a fictitous boba shop. This app incorporates working cart/checkout process and integration of Stripe mobile payments.
                    </p>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      React
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Redux
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Styled Components
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Express
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      JWT
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Stripe-js
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      MongoDB
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Netlify
                    </button>
                  </div>
                  <div
                    class="card__footer"
                  >
                    <a
                      href="https://bobalife.netlify.app"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      Live Site
                    </a>
                    <a
                      href="https://github.com/danveb/bobalife"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      GitHub Repo
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="card__container"
              >
                <div
                  class="image__container"
                >
                  <img
                    alt="portfolio project showcase"
                    src="asador.png"
                  />
                </div>
                <div
                  class="card__content"
                >
                  <div
                    class="card__title"
                  >
                    <h1>
                      asador
                    </h1>
                  </div>
                  <div
                    class="card__body"
                  >
                    <p>
                      Ever wondered about asado (barbeque) in Buenos Aires? Asador is all about sharing the joy of the best parrillas (steakhouses). Users are greeted with a live map with pins. Signing up for an account allows users to pin their favorite parrillas.
                    </p>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      React
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Redux
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Sass
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Express
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      JWT
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      MapboxGL API
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      MongoDB
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Netlify
                    </button>
                  </div>
                  <div
                    class="card__footer"
                  >
                    <a
                      href="https://asador.netlify.app"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      Live Site
                    </a>
                    <a
                      href="https://github.com/danveb/asador.ly"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      GitHub Repo
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="card__container"
              >
                <div
                  class="image__container"
                >
                  <img
                    alt="portfolio project showcase"
                    src="ioniq5.png"
                  />
                </div>
                <div
                  class="card__content"
                >
                  <div
                    class="card__title"
                  >
                    <h1>
                      ioniq5 Lookup
                    </h1>
                  </div>
                  <div
                    class="card__body"
                  >
                    <p>
                      App designed for users who want to search deliveries of 2022 Hyundai IONIQ 5 to local dealerships. User can fill in the zipcode and radius for searching. Data is fetched from Hyundai USA's API and key details are displayed on a table.
                    </p>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      React
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Sass
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      MUI
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Express
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Hyundai USA API
                    </button>
                  </div>
                  <div
                    class="btn"
                  >
                    <button>
                      Netlify
                    </button>
                  </div>
                  <div
                    class="card__footer"
                  >
                    <a
                      href="https://ioniq5-lookup.netlify.app/"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      Live Site
                    </a>
                    <a
                      href="https://github.com/danveb/ioniq5"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      GitHub Repo
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="card__container"
              >
                <div
                  class="image__container"
                >
                  <img
                    alt="portfolio project showcase"
                    src="laptop1.png"
                  />
                </div>
                <div
                  class="card__content"
                >
                  <div
                    class="card__title"
                  >
                    <h1>
                      Check my GitHub
                    </h1>
                  </div>
                  <div
                    class="card__body"
                  >
                    <p>
                      Check my GitHub link to view more projects where I continue to build user-focused apps. Follow the link below!
                    </p>
                  </div>
                  <div
                    class="card__footer"
                  >
                    <a
                      href=""
                      rel="noreferrer noopener"
                      target="_blank"
                    />
                    <a
                      href="https://github.com/danveb"
                      rel="noreferrer noopener"
                      target="_blank"
                    >
                      GitHub Repo
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="contact"
        id="contact"
      >
        <div
          class="contact__wrapper"
        >
          <div
            class="contact__main"
          >
            <div
              class="contact__title"
            >
              <h1>
                Send Me A Note
                <span>
                  .
                </span>
              </h1>
            </div>
            <div
              class="contact__intro"
            >
              <p>
                I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!
              </p>
              <a
                class="cta"
                href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!"
                rel="noreferrer noopener"
                target="_blank"
              >
                Say Hello
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="footer__wrapper"
        >
          <div
            class="footer__title"
          >
            <h1>
              Social Links
              <span>
                .
              </span>
            </h1>
          </div>
          <div
            class="f__socials"
          >
            <ul>
              <li>
                <a
                  href="https://www.github.com/danveb"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="GitHubIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                    />
                  </svg>
                </a>
              </li>
              <li>
                <a
                  href="https://www.linkedin.com/in/daniel-eze-bae"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LinkedInIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                    />
                  </svg>
                </a>
              </li>
              <li>
                <a
                  href="mailto:danibae91@gmail.com"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AlternateEmailIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10h5v-2h-5c-4.34 0-8-3.66-8-8s3.66-8 8-8 8 3.66 8 8v1.43c0 .79-.71 1.57-1.5 1.57s-1.5-.78-1.5-1.57V12c0-2.76-2.24-5-5-5s-5 2.24-5 5 2.24 5 5 5c1.38 0 2.64-.56 3.54-1.47.65.89 1.77 1.47 2.96 1.47 1.97 0 3.5-1.6 3.5-3.57V12c0-5.52-4.48-10-10-10zm0 13c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </div>
          <div
            class="footer__mr"
          >
            <p>
              Portfolio design & development - Danny Bae
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
